fhirpipe/transform/fhir.py

```diff
@@ -0,0 +1,139 @@
+
+    if attribute_structure["mergingScript"]:
+        result = row[
+            new_col_name(attribute_structure["mergingScript"], cols_to_fetch)
+        ]
+    else:
+        result = [row[c] for c in cols_to_fetch[0]]
+        result.extend(cols_to_fetch[1])
+
+    # Unlist
+    if len(result) == 1:
+        result = result[0]
+
+    return result
+
+
+def rec_create_fhir_object(fhir_obj, attribute_structure, row):
+    if isinstance(attribute_structure, dict):
+        # If there are some inputs, we can build the objects to output
+        if (
+            "inputColumns" in attribute_structure.keys()
+            and attribute_structure["inputColumns"]
+        ):
+            result = build_fhir_attribute(attribute_structure, row)
+            fhir_obj[attribute_structure["name"]] = result
+
+        # Otherwise, we can recurse on children
+        else:
+            is_list = attribute_structure["type"].startswith("list")
+
+            fhir_obj[attribute_structure["name"]] = [] if is_list else dict()
+
+            for attr in attribute_structure["attributes"]:
+                child = {} if is_list else fhir_obj[attribute_structure["name"]]
+                rec_create_fhir_object(child, attr, row)
+
+                if is_list:
+                    if attribute_structure["type"] != "list":
+                        child = list(child.values())
+                        if len(child) == 1:
+                            child = child[0]
+
+                    fhir_obj[attribute_structure["name"]].append(child)
+
+            # If the object is a Reference, to we give it to bind_reference
+            if attribute_structure["type"].startswith("Reference"):
+                bind_reference(
+                    fhir_obj[attribute_structure["name"]], attribute_structure
+                )
+
+    # If the current object is a list, we can repeat the same steps as above for each item
+    elif isinstance(fhir_obj, list) and len(fhir_obj) > 0:
+        children = []
+        for child_spec in fhir_obj:
+            children.append(rec_create_fhir_object(dict(), child_spec, row))
+
+        fhir_obj[attribute_structure["name"]] = children
+
+
+def bind_reference(fhir_object, fhir_spec):
+    """
+    Analyse a reference and replace the provided identifier
+    with official FHIR uri if the resource reference exists.
+
+    args:
+        fhir_object: the fhir object to parse
+
+    return:
+        a fhir_object where valid references are now FHIR uris
+    """
+
+    # First we check that the reference has been provided
+    if fhir_object and fhir_object["identifier"]["value"]:
+        identifier = fhir_object["identifier"]["value"]
+
+        # Collect all the resource_types which could be referenced
+        try:
+            begin_ind = fhir_spec["type"].find("(") + 1
+            end_ind = fhir_spec["type"].find(")")
+            resource_types = fhir_spec["type"][begin_ind:end_ind].split("|")
```

Comment:
A compact version:
resource_types  = re.search(r'\((.*?)\)', fhir_spec["type"]).group(1).split("|")

Reply:
As discussed, we unfortunately don't have this info anymore :(

Reply:
yeah! :(